clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java

```diff
@@ -122,6 +122,10 @@ public int timeout() {
         return partitionRecords;
     }
 
+    public void emptyData() {
```

Comment:
I know this name was used in the old `ProduceRequest`, but it seems a bit ambiguous to me. Something with `clear` in the name would be clearer for me (`clearData`, `clearRecords`, `clearPartitionRecords`, etc.).